File: Easy/ValidParentheses/ValidParentheses.java
package ValidParentheses;
//https://leetcode.com/problems/valid-parentheses/

class Solution {
    public boolean isValid(String s) {
        Stack<Character> brackets = new Stack<Character>();
        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == '(') {
                brackets.push(')');
            } else if (s.charAt(i) == '[') {
                brackets.push(']');
            } else if (s.charAt(i) == '{') {
                brackets.push('}');
            } else if (brackets.empty() || s.charAt(i) != brackets.pop()) {
                return false;
            }
        }

        return brackets.empty();
    }
}

File: Easy/ValidParentheses/ValidParentheses.py
# https://leetcode.com/problems/valid-parentheses/

class Solution:
    def isValid(self, s: str) -> bool:
        stack = deque()
        map={'(':')', '[':']', '{':'}'}

        for i in range(0, len(s)):
            if s[i]=='(' or s[i]=='[' or s[i]=='{':
                stack.append(map[s[i]])
            elif len(stack)==0 or stack.pop()!=s[i]:
                return False
        
        return len(stack)==0
